### ISSUE TITLE
output wrong predict in ios 11 device

### ISSUE THREAD

**Issue by fcu-d0515211** [tf-coreml/tf-coreml#219]
I had successful convert .pb to .mlmodel format
In python i had test the predict from mlmodel
But in ios device it keep return result 0.03 / 0.04 (after softmax)
![image](https://user-images.githubusercontent.com/22321340/42736773-16804b44-889e-11e8-8ba0-1d9b325d1085.png)

**Comment by fcu-d0515211** [comment]
This is my coreml_converter
![image](https://user-images.githubusercontent.com/22321340/42924834-88461b60-8b5e-11e8-9a16-d33422260b97.png)
I found that in ipad work fine, but iphone get the 0.03

This is my load model code and predict
`
    func initialModel() {
        guard let model = try? VNCoreMLModel(for: EnglishTextRecognition().model) else {
            fatalError("can't load Places ML model")
        }
        print("initial model")
        self.model = model
    }
    
    fileprivate func detectImage(Image image: UIImage) {
        // Load the ML model through its generated class
        if model == nil {
            initialModel()
        }
        let cgSize = CGSize(width: 128, height: 128)
        let resizedImage = image.resizeImage(targetSize: cgSize)
        let cgImage = resizedImage.cgImage
        
        // Create a Vision request with completion handler
        let request = VNCoreMLRequest(model: model) { [weak self] request, error in
            print("request")
            DispatchQueue.main.async {
                guard let results = request.results else {
                    print("Unable to classify image.\n\(error!.localizedDescription)")
                    self?.dispatchGroup.leave()
                    return
                }
                // The `results` will always be `VNClassificationObservation`s, as specified by the Core ML model in this project.
                let classifications = results as! [VNClassificationObservation]
                
                if classifications.isEmpty {
                    self?.recognizedText = "Nothing recognized."
                    print("Nothing recognized.")
                } else {
                    // Display top classifications ranked by confidence in the UI.
                    let topClassifications = classifications.first!
                    let descriptions = String(format: "  (%.2f) %@", topClassifications.confidence, topClassifications.identifier)
                    let text = self?.recognizedText
                    self?.recognizedText = text! + topClassifications.identifier
                }
                self?.dispatchGroup.leave()
            }
        }
        `

**Comment by aseemw** [comment]
If its working in python and giving the expected (i.e. same as TF) output with coremltools predict, then maybe the input in the app is different numerically than the the one that you are using in python. 

Whats the code you used to verify the correctness in python? 
Do you get the same predictions if you make a mac app? (sometimes predictions from Mac and iOS can differ, if that happens its a bug).

**Comment by fcu-d0515211** [comment]
i use the giving code in examples to verify the correctness and also got doing some image preprocessing for tensorflow . 
if it is a bug, how can i fix it? or have any recommend tricks to avoid it?

**Comment by aseemw** [comment]
It could be an issue with the preprocessing parameters not set properly during conversion. Pleas take a look at this [example](https://github.com/tf-coreml/tf-coreml/blob/master/examples/inception_v1_preprocessing_steps.ipynb)

**Comment by fcu-d0515211** [comment]
I had follow the example. The model is successful be converted and verify perfectly by using bottom code.
```

import tensorflow as tf
import  numpy as np
import PIL.Image as Image
import PIL
from skimage import io, transform
import sys
import coremltools
from coremltools.models import datatypes, utils, MLModel

labels = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z', 'a', 'b', 'c', 'comma', 'd', 'dash', 'dot', 'e', 'exclaimation', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'question', 'r', 's', 'space', 't', 'u', 'underscore', 'v', 'w', 'x', 'y', 'z']

def recognize_tf(jpg_path, pb_file_path):
  with tf.Graph().as_default():
    output_graph_def = tf.GraphDef()

    with open(pb_file_path, "rb") as f:
      output_graph_def.ParseFromString(f.read())
      _ = tf.import_graph_def(output_graph_def, name="")

    with tf.Session() as sess:
      init = tf.global_variables_initializer()
      sess.run(init)

      input_x = sess.graph.get_tensor_by_name("input_x:0")
      print(input_x)
      out_softmax = sess.graph.get_tensor_by_name("Output_Layer/softmax_pred:0")
      print(out_softmax)
      # out_label = sess.graph.get_tensor_by_name("output_y:0")
      # print(out_label)

      img = io.imread(jpg_path)
      img = transform.resize(img, (224, 224, 3))
      img_out_softmax = sess.run(out_softmax, feed_dict={input_x: np.reshape(img, [-1, 224, 224, 3])})

      print("img_out_softmax:", img_out_softmax)
      prediction = np.argmax(img_out_softmax, axis=1)
      print("index:", prediction)
      print("label:", labels[int(prediction)])

def recognize_coreml(jpg_path, mlmodel_path):
  spec = utils.load_spec(mlmodel_path)
  coreml_model = MLModel(spec)
  img = Image.open(jpg_path)
  img = img.resize([224,224], PIL.Image.ANTIALIAS)

  coreml_inputs = {'input_x__0': img}
  coreml_output = coreml_model.predict(coreml_inputs, useCPUOnly=True)
  coreml_pred_dict = coreml_output['Output_Layer__softmax_pred__0']
  coreml_predicted_class_label = coreml_output['classLabel']

  print("CoreML prediction class = {}, probabiltiy = {}".format(coreml_predicted_class_label, str(coreml_pred_dict)))

folder = "tf9"
pb_name = "result"
coreml_name = "EnglishTextRecognition"

recognize_tf(sys.argv[1], "{}/{}.pb".format(folder, pb_name))
recognize_coreml(sys.argv[1], '{}/{}.mlmodel'.format(folder, coreml_name))
```
and the result is this correctly
<img width="829" alt="screen shot 2018-07-21 at 4 54 16 am" src="https://user-images.githubusercontent.com/22321340/43024923-687dd6f2-8ca2-11e8-96c6-49575733b015.png">

But when I put it into my ios device iphone 7 and run with vision(crop word from image) and core ml to predict. it get low result.

**Comment by fcu-d0515211** [comment]
I found i had this error. What this mean?
Shapes not found for 37 tensors. Executing graph to determine shapes.
![image](https://user-images.githubusercontent.com/22321340/43038878-629e7f04-8d54-11e8-9762-a351e0f8362a.png)

**Comment by Mstronach** [comment]
Hi, are you still experiencing this issue?
